Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/Settings.java

```diff
@@ -87,6 +98,70 @@ public boolean isGameSpecific()
     return !TextUtils.isEmpty(mGameId);
   }
 
+  public IniFile getWiimoteProfile(String profile, int padID)
+  {
+    File wiimoteProfileFile = SettingsFile.getWiiProfile(profile);
+    IniFile wiimoteProfileIni = new IniFile();
+
+    if (!wiimoteProfileIni.load(wiimoteProfileFile, false))
+    {
+      String defaultWiiProfilePath = DirectoryInitialization.getUserDirectory() +
+              "/Config/Profiles/Wiimote/WiimoteProfile.ini";
+
+      wiimoteProfileIni.load(defaultWiiProfilePath, false);
+
+      wiimoteProfileIni.setString(Settings.SECTION_PROFILE, "Device",
+              "Android/" + (padID + 4) + "/Touchscreen");
+    }
+
+    switch (padID)
+    {
+      case 0:
+        mWiimoteProfileFile0 = wiimoteProfileFile;
+        mWiimoteProfileIni0 = wiimoteProfileIni;
+        return mWiimoteProfileIni0;
+
+      case 1:
+        mWiimoteProfileFile1 = wiimoteProfileFile;
+        mWiimoteProfileIni1 = wiimoteProfileIni;
+        return mWiimoteProfileIni1;
+
+      case 2:
+        mWiimoteProfileFile2 = wiimoteProfileFile;
+        mWiimoteProfileIni2 = wiimoteProfileIni;
+        return mWiimoteProfileIni2;
+
+      case 3:
+        mWiimoteProfileFile3 = wiimoteProfileFile;
+        mWiimoteProfileIni3 = wiimoteProfileIni;
+        return mWiimoteProfileIni3;
+
+      default:
+        throw new IllegalStateException("Invalid padID");
+    }
+  }
+
+  public void enableWiimoteProfile(Settings settings, String profile, String profileKey)
+  {
+    getWiimoteControlsSection(settings).setString(profileKey, profile);
+  }
+
+  public boolean disableWiimoteProfile(Settings settings, String profileKey)
+  {
+    return getWiimoteControlsSection(settings).delete(profileKey);
+  }
+
+  public boolean isWiimoteProfileEnabled(Settings settings, String profile,
+          String profileKey)
+  {
+    return profile.equals(getWiimoteControlsSection(settings).getString(profileKey, ""));
+  }
+
+  public IniFile.Section getWiimoteControlsSection(Settings s)
+  {
+    return s.getSection(GAME_SETTINGS_PLACEHOLDER_FILE_NAME, Settings.SECTION_CONTROLS);
```

Comment:
If you're going to drop the `Settings` prefix from `GAME_SETTINGS_PLACEHOLDER_FILE_NAME`, you should drop it from `SECTION_CONTROLS` too. (And you could also rename `s` to `settings`, because I only did that to make this fit on one line,